{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "9a23a31c-585d-4414-98ff-63b52bae07bb",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "# Try psm format"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "cebe4394-ef55-4028-b252-4085a90504ab",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['FileName', 'ScanNum', 'QueryPrecursorMz', 'OriginalPrecursorMz',\n",
      "       'PrecursorError(ppm)', 'QueryCharge', 'OriginalCharge',\n",
      "       'PrecursorScanNum', 'PrecursorArea', 'PrecursorRelAb',\n",
      "       'RTAtPrecursorHalfElution', 'PeptideSequence', 'AmbiguousMatch',\n",
      "       'Protein', 'DeNovoScore', 'MSGFScore', 'Evalue', 'Qvalue', 'PepQvalue',\n",
      "       'PrecursorPurity', 'FractionDecomposition', 'HCDEnergy', 'iTRAQ114',\n",
      "       'iTRAQ115', 'iTRAQ116', 'iTRAQ117', 'iTRAQFlags', 'iTRAQTotalAb',\n",
      "       'iTRAQFractionOfTotalAb'],\n",
      "      dtype='object')\n",
      "                                                    FileName   ScanNum  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...       365  343.5114   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...      2129  347.9633   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...      2132  494.6088   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...      2144  391.5446   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...      2155  561.3084   \n",
      "\n",
      "                                                    QueryPrecursorMz  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...          343.5114   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...          347.9633   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...          494.6088   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...          391.5446   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...          561.3084   \n",
      "\n",
      "                                                    OriginalPrecursorMz  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...             5.330432   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...             7.279439   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...             3.702040   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...             4.364744   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...             4.893202   \n",
      "\n",
      "                                                    PrecursorError(ppm)  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...                    3   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...                    4   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...                    3   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...                    3   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...                    2   \n",
      "\n",
      "                                                    QueryCharge  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...            3   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...            4   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...            3   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...            3   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...            2   \n",
      "\n",
      "                                                   OriginalCharge  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...            364   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...          2127?   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...           2131   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...          2142?   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...           2154   \n",
      "\n",
      "                                                    PrecursorScanNum  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...               0.0   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...        30134200.0   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...               0.0   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...        93147400.0   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...               0.0   \n",
      "\n",
      "                                                    PrecursorArea  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...       0.000000   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...       0.000007   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...       0.000000   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...       0.000020   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...       0.000000   \n",
      "\n",
      "                                                    PrecursorRelAb  ...  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...               0  ...   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...             656  ...   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...               0  ...   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...             657  ...   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...               0  ...   \n",
      "\n",
      "                                                   PrecursorPurity  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...     100.0,100.0   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...       86.3,87.1   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...       95.6,96.1   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...       94.2,94.2   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...       80.3,78.6   \n",
      "\n",
      "                                                    FractionDecomposition  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...     16.3511352539063eV   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...     15.5887289047241eV   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...     23.5433979034424eV   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...     18.6375026702881eV   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...     28.2900848388672eV   \n",
      "\n",
      "                                                        HCDEnergy  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...  8984.92/-0.03   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...   56511.7/0.06   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...   40574.7/0.05   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...    333609/0.07   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...   61306.9/0.09   \n",
      "\n",
      "                                                        iTRAQ114  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...   5003.1/0.02   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...  58536.3/0.05   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...  46578.5/0.08   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...   201102/0.05   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...  16606.1/0.04   \n",
      "\n",
      "                                                        iTRAQ115  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...  12337.2/0.06   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...  40089.4/0.06   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...  29148.5/0.06   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...   365177/0.07   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...  25922.1/0.04   \n",
      "\n",
      "                                                         iTRAQ116  iTRAQ117  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...  1504.79/-0.00         I   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...     32863/0.09       NaN   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...  7378.54/-0.02       NaN   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...   82645.5/0.16         I   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...   57662.5/0.08       NaN   \n",
      "\n",
      "                                                    iTRAQFlags iTRAQTotalAb  \\\n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...     27830.0     0.129261   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...    188000.0     0.206374   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...    123680.0     0.301408   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...    982534.0     0.177074   \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...    161498.0     0.314821   \n",
      "\n",
      "                                                   iTRAQFractionOfTotalAb  \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...                    NaN  \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...                    NaN  \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...                    NaN  \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...                    NaN  \n",
      "TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024...                    NaN  \n",
      "\n",
      "[5 rows x 29 columns]\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "psm_df = pd.read_csv(\"./dataset/TCGA/Proteomics/Downloads/TCGA_A2-A0SW_AO-A0JL_BH-A0BV_117C_W_BI_20131024_H-PM_f16.raw.cap.psm\", sep=\"\\t\")\n",
    "\n",
    "print(psm_df.columns)\n",
    "print(psm_df.head())\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e28635ba-e155-4e02-a930-ca093f6cc97e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Apply filtering\n",
    "filtered_psm_df = psm_df[\n",
    "    (psm_df[\"PrecursorError(ppm)\"] < 10) &\n",
    "    (psm_df[\"Qvalue\"] < 0.01) &\n",
    "    (psm_df[\"PepQvalue\"] < 0.01) &\n",
    "    (psm_df[\"PrecursorPurity\"] > 90) &\n",
    "    (psm_df[\"AmbiguousMatch\"] == 0)  # Keep only unambiguous matches\n",
    "]\n",
    "\n",
    "# Example: Grouping peptides by protein\n",
    "peptide_to_protein = filtered_psm_df.groupby(\"Protein\")[\"PeptideSequence\"].apply(set)\n",
    "\n",
    "\n",
    "# Group by Protein and compute median iTRAQ intensities\n",
    "protein_expression_df = filtered_psm_df.groupby(\"Protein\")[[\"iTRAQ114\", \"iTRAQ115\", \"iTRAQ116\", \"iTRAQ117\"]].median()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "760ce5a8-9a70-4776-9bd8-2af19388d0d0",
   "metadata": {},
   "source": [
    "# Try "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "1f76ec36-fe48-440d-b002-a23c1b3f3881",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ðŸ“¥ Download Progress: 100%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ| 121/121 [00:00<00:00, 16208.19file/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "âœ… 29ec889a-6fef-40d9-9a6c-570e61c25d0b.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… b7e3fbe6-1ece-4bbb-9d50-222daafe0e34.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… a9972ef0-7002-463a-bc65-c59c6d50ac59.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… b29b00ab-501a-4797-babc-6628eba5f501.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… edeeefbe-8d5f-46a2-a75c-549688de51ae.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 8dabfed6-1eb1-4915-8203-66093d2cf43f.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 238ce745-a5f9-4f2e-a9c6-4e353d72bbeb.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 136e782c-cfa4-450a-8ed7-55a1d610c12c.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 085415a5-37c0-456e-89aa-2c7c4e1230ab.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… ad3dbcd4-b766-47c0-8c57-007392299aa0.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 5619c1e7-4b0d-4815-8538-fd64cecd50d8.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 749da78c-968f-4682-a2fa-5c82ce35a98d.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… a474e5eb-bd87-4a5f-91aa-d3c350ec44d5.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 9c38cc51-ac3b-4ae0-afe3-4d76f001ae8e.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 7b7eb959-7d58-48bc-8da3-6e4dad9a8ac5.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 0f28a25e-ba92-4279-869d-b641e2f5c72b.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… c18e524d-4a68-4fc2-9a21-83ee8a762aa8.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… b2535571-da19-413f-a979-5c388399e619.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… dddabbdc-a8b0-4571-820d-ed39781cebed.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 435866be-7544-48cc-954f-17748e1580c0.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… c5b72fde-2cf0-4dbc-8cde-41840080d542.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 64e4c231-1793-44a8-8e80-b050658b42d6.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… f3f750a9-5492-4d70-927a-0b091fea2f69.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 4dc10240-c4ca-4b92-866b-af1542575fc8.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 3692f9df-5b70-4cfb-a504-ed2cb4164ac3.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 5f173a3c-89f0-41c9-bbd8-5256a0489d4e.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… ea6ae785-68fc-435e-8c7b-c09682c018d3.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 49379b08-1e67-4666-88d6-2791428b9d62.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 1783266a-bcae-4f80-856c-dafc23b65006.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 17c51f0c-6347-4fb4-ae8c-e7933b605b7e.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 89b5ed3c-3a44-4e0b-8258-02290c6c5abe.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… fcdc30f9-f2d0-4fe9-93f7-fe3ca91e826c.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 7d207f6d-7de9-40db-8f06-0ba1ce1314e2.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 4ce4d688-ce22-43bf-817b-393ee751da34.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 8256500e-ea1f-487e-9b69-7e6f6c992334.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… ba0570bd-508d-4b93-b147-365af6a86241.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 4ebec7e0-8377-4c46-b153-d7b084aa7cab.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 2d5c7af9-63ae-4461-847a-79d379cb59f6.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 116b9b65-ffc3-4cd6-8a08-aa3e28f76269.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… fb9b328b-eaa7-4372-9444-fc75fbf8a0c5.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… b1aeced1-1b77-41ac-92ee-189f94c1650f.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… f6591087-c0c2-4afb-949e-cba3689cb3ca.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 4b987a90-4779-4930-9206-69ea483acbac.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… b5b99dc0-b9e7-4b80-89cd-f42e9b2ed710.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… f18e4f29-844f-4cec-ad36-0ebe9491ed3d.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… de6ffb0a-19a8-46c2-9a45-1e2ea082ee86.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 06c5f033-20a4-4829-b832-573bafc9b80c.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… d43d749c-eea4-4467-a1c9-751923cefbc6.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… a739e3ff-f172-4f9c-94b6-b9fcc4458627.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… fc6e0c8a-46a8-4cfc-b824-8181a8984072.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… f1c423bc-6155-4cfb-8cc8-a98ccd97bd2e.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… b89bc06f-91ca-4dd7-834d-87bb730048cd.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 4f1c4f93-7ce5-4348-82b4-20bf6bd178c4.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n",
      "âœ… 4a9a8be4-1c11-4c8b-b943-77e122513731.rna_seq.augmented_star_gene_counts.tsv already exists. Skipping download.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import requests\n",
    "import os\n",
    "from tqdm import tqdm  # Progress bar\n",
    "\n",
    "# Load the case-to-file mapping data\n",
    "matched_df = pd.read_csv(\"dataset/processed/1. Data_preparation-matching_ID/Proteome/matched_samples_with_aliquot.csv\")\n",
    "gdc_df = pd.read_csv(\"dataset/CPTAC/RNA-Seq/gdc_sample_sheet.2025-02-14.tsv\", sep=\"\\t\")\n",
    "\n",
    "# Ensure Case IDs are strings to prevent type mismatch\n",
    "matched_samples = set(matched_df[\"Case Submitter ID\"].astype(str))\n",
    "\n",
    "# Create a mapping of Case ID â†’ (File ID, File Name)\n",
    "case_to_file = {\n",
    "    row[\"Case ID\"]: (row[\"File ID\"], row[\"File Name\"])\n",
    "    for _, row in gdc_df.iterrows() if row[\"Case ID\"] in matched_samples\n",
    "}\n",
    "\n",
    "# Convert to DataFrame for easier visualization\n",
    "case_to_file_df = pd.DataFrame(\n",
    "    [(case, file_id, file_name) for case, (file_id, file_name) in case_to_file.items()],\n",
    "    columns=[\"Case ID\", \"File ID\", \"File Name\"]\n",
    ")\n",
    "\n",
    "# Create a directory to store downloaded files\n",
    "download_dir = \"dataset/processed/1. Data_preparation-matching_ID/RNA-seq\"\n",
    "os.makedirs(download_dir, exist_ok=True)\n",
    "\n",
    "# GDC API endpoint\n",
    "gdc_api_url = \"https://api.gdc.cancer.gov/data/{}\"\n",
    "\n",
    "# ðŸ”¹ 2ï¸âƒ£ Batch download files with tqdm progress bar\n",
    "for idx, row in tqdm(case_to_file_df.iterrows(), total=len(case_to_file_df), desc=\"ðŸ“¥ Download Progress\", unit=\"file\"):\n",
    "    file_id = row[\"File ID\"]\n",
    "    file_name = row[\"File Name\"]\n",
    "    file_path = os.path.join(download_dir, file_name)\n",
    "\n",
    "    # Check if the file already exists\n",
    "    if os.path.exists(file_path):\n",
    "        print(f\"âœ… {file_name} already exists. Skipping download.\")\n",
    "        continue  # Skip to the next file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "78506fef-5831-4c81-861f-e780a2db0666",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Data type</th>\n",
       "      <th>Available sources</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CNV</td>\n",
       "      <td>[bcm, washu]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>miRNA</td>\n",
       "      <td>[bcm]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>phosphoproteomics</td>\n",
       "      <td>[bcm, umich]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>proteomics</td>\n",
       "      <td>[bcm, umich]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>transcriptomics</td>\n",
       "      <td>[bcm, broad, washu]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ancestry_prediction</td>\n",
       "      <td>[harmonized]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>somatic_mutation</td>\n",
       "      <td>[harmonized, washu]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>clinical</td>\n",
       "      <td>[mssm]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>follow-up</td>\n",
       "      <td>[mssm]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>medical_history</td>\n",
       "      <td>[mssm]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>acetylproteomics</td>\n",
       "      <td>[umich]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>cibersort</td>\n",
       "      <td>[washu]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>hla_typing</td>\n",
       "      <td>[washu]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>tumor_purity</td>\n",
       "      <td>[washu]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>xcell</td>\n",
       "      <td>[washu]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Data type    Available sources\n",
       "0                   CNV         [bcm, washu]\n",
       "1                 miRNA                [bcm]\n",
       "2     phosphoproteomics         [bcm, umich]\n",
       "3            proteomics         [bcm, umich]\n",
       "4       transcriptomics  [bcm, broad, washu]\n",
       "5   ancestry_prediction         [harmonized]\n",
       "6      somatic_mutation  [harmonized, washu]\n",
       "7              clinical               [mssm]\n",
       "8             follow-up               [mssm]\n",
       "9       medical_history               [mssm]\n",
       "10     acetylproteomics              [umich]\n",
       "11            cibersort              [washu]\n",
       "12           hla_typing              [washu]\n",
       "13         tumor_purity              [washu]\n",
       "14                xcell              [washu]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "en = cptac.cptac.Brca()\n",
    "en.list_data_sources()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "2e434bb7-f853-40d1-9270-1f4a955d68cd",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Downloading Report_abundance_groupby=protein_protNorm=MD_gu=2.tsv.gz: 100%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ| 15.6M/15.6M [00:04<00:00, 3.50MB/s]\n",
      "Downloading prosp-brca-all-samples.txt.gz: 100%|â–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆâ–ˆ| 3.85k/3.85k [00:00<00:00, 19.5kB/s]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Samples: ['01BR001', '01BR008', '01BR009', '01BR010', '01BR015', '01BR017', '01BR018', '01BR020', '01BR023', '01BR025', '01BR026', '01BR027', '01BR030', '01BR031', '01BR032', '01BR033', '01BR040', '01BR042', '01BR043', '03BR002']\n",
      "Proteins: [('ARF5', 'ENSP00000000233.5'), ('M6PR', 'ENSP00000000412.3'), ('ESRRA', 'ENSP00000000442.6'), ('FKBP4', 'ENSP00000001008.4'), ('NDUFAF7', 'ENSP00000002125.4'), ('FUCA2', 'ENSP00000002165.5'), ('DBNDD1', 'ENSP00000002501.6'), ('SEMA3F', 'ENSP00000002829.3'), ('CFTR', 'ENSP00000003084.6'), ('CYP51A1', 'ENSP00000003100.8'), ('USP28', 'ENSP00000003302.4'), ('TMEM176A', 'ENSP00000004103.3'), ('SLC7A2', 'ENSP00000004531.10'), ('HSPB6', 'ENSP00000004982.3'), ('ZNF195', 'ENSP00000005082.9'), ('PDK4', 'ENSP00000005178.5'), ('RALA', 'ENSP00000005257.2'), ('BAIAP2L1', 'ENSP00000005260.8'), ('TMEM132A', 'ENSP00000005286.4'), ('DVL2', 'ENSP00000005340.4')]\n"
     ]
    }
   ],
   "source": [
    "# there is also a get function for each datatype, this is the same as get_dataframe \n",
    "# but does not need the datatype argument. This function call below is equivalent to \n",
    "# the one above\n",
    "proteomics = en.get_proteomics('umich')\n",
    "\n",
    "samples = proteomics.index\n",
    "proteins = proteomics.columns\n",
    "print(\"Samples:\",samples[0:20].tolist()) #the first twenty samples\n",
    "print(\"Proteins:\",proteins[0:20].tolist()) #the first twenty proteins"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "46614767-1aa3-494a-ae9a-4467dbf8e454",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th>ARF5</th>\n",
       "      <th>M6PR</th>\n",
       "      <th>ESRRA</th>\n",
       "      <th>FKBP4</th>\n",
       "      <th>NDUFAF7</th>\n",
       "      <th>FUCA2</th>\n",
       "      <th>DBNDD1</th>\n",
       "      <th>SEMA3F</th>\n",
       "      <th>CFTR</th>\n",
       "      <th>CYP51A1</th>\n",
       "      <th>...</th>\n",
       "      <th>DDHD1</th>\n",
       "      <th>WIZ</th>\n",
       "      <th>GBF1</th>\n",
       "      <th>APOA5</th>\n",
       "      <th>WIZ</th>\n",
       "      <th>LDB1</th>\n",
       "      <th>WIZ</th>\n",
       "      <th>RFX7</th>\n",
       "      <th>SWSAP1</th>\n",
       "      <th>SVIL</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Database_ID</th>\n",
       "      <th>ENSP00000000233.5</th>\n",
       "      <th>ENSP00000000412.3</th>\n",
       "      <th>ENSP00000000442.6</th>\n",
       "      <th>ENSP00000001008.4</th>\n",
       "      <th>ENSP00000002125.4</th>\n",
       "      <th>ENSP00000002165.5</th>\n",
       "      <th>ENSP00000002501.6</th>\n",
       "      <th>ENSP00000002829.3</th>\n",
       "      <th>ENSP00000003084.6</th>\n",
       "      <th>ENSP00000003100.8</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSP00000500986.2</th>\n",
       "      <th>ENSP00000500993.1</th>\n",
       "      <th>ENSP00000501064.1</th>\n",
       "      <th>ENSP00000501141.1</th>\n",
       "      <th>ENSP00000501256.3</th>\n",
       "      <th>ENSP00000501277.1</th>\n",
       "      <th>ENSP00000501300.1</th>\n",
       "      <th>ENSP00000501317.1</th>\n",
       "      <th>ENSP00000501355.1</th>\n",
       "      <th>ENSP00000501521.1</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>01BR001</th>\n",
       "      <td>0.012367</td>\n",
       "      <td>-0.945999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.478170</td>\n",
       "      <td>1.135840</td>\n",
       "      <td>-0.512706</td>\n",
       "      <td>0.750335</td>\n",
       "      <td>-0.274824</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.278244</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.649127</td>\n",
       "      <td>-0.580869</td>\n",
       "      <td>-0.226667</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.676185</td>\n",
       "      <td>-0.068202</td>\n",
       "      <td>-0.078207</td>\n",
       "      <td>-0.328420</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR008</th>\n",
       "      <td>-0.514386</td>\n",
       "      <td>0.462307</td>\n",
       "      <td>0.230124</td>\n",
       "      <td>-0.555968</td>\n",
       "      <td>0.491366</td>\n",
       "      <td>-0.656034</td>\n",
       "      <td>-1.220890</td>\n",
       "      <td>-0.369282</td>\n",
       "      <td>-1.036441</td>\n",
       "      <td>-0.059327</td>\n",
       "      <td>...</td>\n",
       "      <td>0.632221</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.032873</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.015459</td>\n",
       "      <td>0.227424</td>\n",
       "      <td>0.325643</td>\n",
       "      <td>-0.606240</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR009</th>\n",
       "      <td>-0.210782</td>\n",
       "      <td>-0.085055</td>\n",
       "      <td>0.380296</td>\n",
       "      <td>-0.389491</td>\n",
       "      <td>1.255391</td>\n",
       "      <td>-0.608007</td>\n",
       "      <td>-0.231318</td>\n",
       "      <td>0.092870</td>\n",
       "      <td>-1.505195</td>\n",
       "      <td>0.206595</td>\n",
       "      <td>...</td>\n",
       "      <td>0.450818</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.341503</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.220239</td>\n",
       "      <td>0.125092</td>\n",
       "      <td>0.365397</td>\n",
       "      <td>-0.167392</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR010</th>\n",
       "      <td>0.105457</td>\n",
       "      <td>0.351335</td>\n",
       "      <td>-0.322798</td>\n",
       "      <td>-0.821610</td>\n",
       "      <td>0.241406</td>\n",
       "      <td>-0.500140</td>\n",
       "      <td>-0.137824</td>\n",
       "      <td>0.113791</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.498314</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.423470</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.360900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.451556</td>\n",
       "      <td>-0.098897</td>\n",
       "      <td>0.208643</td>\n",
       "      <td>-0.729096</td>\n",
       "      <td>0.670307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>01BR015</th>\n",
       "      <td>-0.509298</td>\n",
       "      <td>-0.874164</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.113804</td>\n",
       "      <td>-0.131347</td>\n",
       "      <td>-0.412813</td>\n",
       "      <td>0.262210</td>\n",
       "      <td>0.042333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.657666</td>\n",
       "      <td>...</td>\n",
       "      <td>0.406016</td>\n",
       "      <td>-0.493869</td>\n",
       "      <td>-0.192847</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.083639</td>\n",
       "      <td>0.966976</td>\n",
       "      <td>-0.012664</td>\n",
       "      <td>0.081968</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21BR010</th>\n",
       "      <td>0.528298</td>\n",
       "      <td>-0.127929</td>\n",
       "      <td>-0.497360</td>\n",
       "      <td>-0.151022</td>\n",
       "      <td>0.082288</td>\n",
       "      <td>0.447267</td>\n",
       "      <td>0.151024</td>\n",
       "      <td>0.220194</td>\n",
       "      <td>0.516739</td>\n",
       "      <td>-0.230357</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.172151</td>\n",
       "      <td>0.636608</td>\n",
       "      <td>0.267400</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.09507</td>\n",
       "      <td>-0.017522</td>\n",
       "      <td>-0.220463</td>\n",
       "      <td>-0.067717</td>\n",
       "      <td>-0.311446</td>\n",
       "      <td>0.602422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22BR005</th>\n",
       "      <td>-0.549542</td>\n",
       "      <td>0.134236</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.580773</td>\n",
       "      <td>-0.080663</td>\n",
       "      <td>-0.056509</td>\n",
       "      <td>-0.148632</td>\n",
       "      <td>0.260986</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.348578</td>\n",
       "      <td>...</td>\n",
       "      <td>0.791937</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.171712</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.083980</td>\n",
       "      <td>-0.200083</td>\n",
       "      <td>0.155198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.456801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22BR006</th>\n",
       "      <td>0.336092</td>\n",
       "      <td>0.125742</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.360510</td>\n",
       "      <td>0.086199</td>\n",
       "      <td>0.470607</td>\n",
       "      <td>-0.515990</td>\n",
       "      <td>-0.162247</td>\n",
       "      <td>1.003075</td>\n",
       "      <td>0.342987</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.080755</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.174904</td>\n",
       "      <td>-0.353412</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.013793</td>\n",
       "      <td>-0.253829</td>\n",
       "      <td>-0.117960</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.094966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CPT000814</th>\n",
       "      <td>-0.518995</td>\n",
       "      <td>0.262582</td>\n",
       "      <td>0.277980</td>\n",
       "      <td>0.137505</td>\n",
       "      <td>0.600041</td>\n",
       "      <td>-1.041230</td>\n",
       "      <td>0.513974</td>\n",
       "      <td>-0.012011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.411714</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.011008</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.035445</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.385942</td>\n",
       "      <td>0.620827</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CPT001846</th>\n",
       "      <td>0.652513</td>\n",
       "      <td>-0.165268</td>\n",
       "      <td>0.809801</td>\n",
       "      <td>-1.108263</td>\n",
       "      <td>0.557576</td>\n",
       "      <td>0.007854</td>\n",
       "      <td>0.213734</td>\n",
       "      <td>-0.721577</td>\n",
       "      <td>-2.695651</td>\n",
       "      <td>0.296581</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.057923</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.252335</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.040722</td>\n",
       "      <td>-0.006118</td>\n",
       "      <td>-0.251360</td>\n",
       "      <td>0.443203</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>125 rows Ã— 12922 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                     ARF5              M6PR             ESRRA  \\\n",
       "Database_ID ENSP00000000233.5 ENSP00000000412.3 ENSP00000000442.6   \n",
       "Patient_ID                                                          \n",
       "01BR001              0.012367         -0.945999               NaN   \n",
       "01BR008             -0.514386          0.462307          0.230124   \n",
       "01BR009             -0.210782         -0.085055          0.380296   \n",
       "01BR010              0.105457          0.351335         -0.322798   \n",
       "01BR015             -0.509298         -0.874164               NaN   \n",
       "...                       ...               ...               ...   \n",
       "21BR010              0.528298         -0.127929         -0.497360   \n",
       "22BR005             -0.549542          0.134236               NaN   \n",
       "22BR006              0.336092          0.125742               NaN   \n",
       "CPT000814           -0.518995          0.262582          0.277980   \n",
       "CPT001846            0.652513         -0.165268          0.809801   \n",
       "\n",
       "Name                    FKBP4           NDUFAF7             FUCA2  \\\n",
       "Database_ID ENSP00000001008.4 ENSP00000002125.4 ENSP00000002165.5   \n",
       "Patient_ID                                                          \n",
       "01BR001             -0.478170          1.135840         -0.512706   \n",
       "01BR008             -0.555968          0.491366         -0.656034   \n",
       "01BR009             -0.389491          1.255391         -0.608007   \n",
       "01BR010             -0.821610          0.241406         -0.500140   \n",
       "01BR015             -0.113804         -0.131347         -0.412813   \n",
       "...                       ...               ...               ...   \n",
       "21BR010             -0.151022          0.082288          0.447267   \n",
       "22BR005              0.580773         -0.080663         -0.056509   \n",
       "22BR006             -0.360510          0.086199          0.470607   \n",
       "CPT000814            0.137505          0.600041         -1.041230   \n",
       "CPT001846           -1.108263          0.557576          0.007854   \n",
       "\n",
       "Name                   DBNDD1            SEMA3F              CFTR  \\\n",
       "Database_ID ENSP00000002501.6 ENSP00000002829.3 ENSP00000003084.6   \n",
       "Patient_ID                                                          \n",
       "01BR001              0.750335         -0.274824               NaN   \n",
       "01BR008             -1.220890         -0.369282         -1.036441   \n",
       "01BR009             -0.231318          0.092870         -1.505195   \n",
       "01BR010             -0.137824          0.113791               NaN   \n",
       "01BR015              0.262210          0.042333               NaN   \n",
       "...                       ...               ...               ...   \n",
       "21BR010              0.151024          0.220194          0.516739   \n",
       "22BR005             -0.148632          0.260986               NaN   \n",
       "22BR006             -0.515990         -0.162247          1.003075   \n",
       "CPT000814            0.513974         -0.012011               NaN   \n",
       "CPT001846            0.213734         -0.721577         -2.695651   \n",
       "\n",
       "Name                  CYP51A1  ...             DDHD1               WIZ  \\\n",
       "Database_ID ENSP00000003100.8  ... ENSP00000500986.2 ENSP00000500993.1   \n",
       "Patient_ID                     ...                                       \n",
       "01BR001             -0.278244  ...         -0.649127         -0.580869   \n",
       "01BR008             -0.059327  ...          0.632221               NaN   \n",
       "01BR009              0.206595  ...          0.450818               NaN   \n",
       "01BR010              0.498314  ...         -0.423470               NaN   \n",
       "01BR015             -0.657666  ...          0.406016         -0.493869   \n",
       "...                       ...  ...               ...               ...   \n",
       "21BR010             -0.230357  ...         -0.172151          0.636608   \n",
       "22BR005             -0.348578  ...          0.791937               NaN   \n",
       "22BR006              0.342987  ...         -0.080755               NaN   \n",
       "CPT000814           -0.411714  ...         -2.011008               NaN   \n",
       "CPT001846            0.296581  ...         -0.057923               NaN   \n",
       "\n",
       "Name                     GBF1             APOA5               WIZ  \\\n",
       "Database_ID ENSP00000501064.1 ENSP00000501141.1 ENSP00000501256.3   \n",
       "Patient_ID                                                          \n",
       "01BR001             -0.226667               NaN               NaN   \n",
       "01BR008              0.032873               NaN               NaN   \n",
       "01BR009             -0.341503               NaN               NaN   \n",
       "01BR010              0.360900               NaN               NaN   \n",
       "01BR015             -0.192847               NaN               NaN   \n",
       "...                       ...               ...               ...   \n",
       "21BR010              0.267400               NaN          -0.09507   \n",
       "22BR005              0.171712               NaN               NaN   \n",
       "22BR006              0.174904         -0.353412               NaN   \n",
       "CPT000814            0.035445               NaN               NaN   \n",
       "CPT001846            0.252335               NaN               NaN   \n",
       "\n",
       "Name                     LDB1               WIZ              RFX7  \\\n",
       "Database_ID ENSP00000501277.1 ENSP00000501300.1 ENSP00000501317.1   \n",
       "Patient_ID                                                          \n",
       "01BR001             -0.676185         -0.068202         -0.078207   \n",
       "01BR008             -0.015459          0.227424          0.325643   \n",
       "01BR009             -0.220239          0.125092          0.365397   \n",
       "01BR010             -0.451556         -0.098897          0.208643   \n",
       "01BR015              0.083639          0.966976         -0.012664   \n",
       "...                       ...               ...               ...   \n",
       "21BR010             -0.017522         -0.220463         -0.067717   \n",
       "22BR005              0.083980         -0.200083          0.155198   \n",
       "22BR006             -0.013793         -0.253829         -0.117960   \n",
       "CPT000814           -1.385942          0.620827               NaN   \n",
       "CPT001846            0.040722         -0.006118         -0.251360   \n",
       "\n",
       "Name                   SWSAP1              SVIL  \n",
       "Database_ID ENSP00000501355.1 ENSP00000501521.1  \n",
       "Patient_ID                                       \n",
       "01BR001             -0.328420               NaN  \n",
       "01BR008             -0.606240               NaN  \n",
       "01BR009             -0.167392               NaN  \n",
       "01BR010             -0.729096          0.670307  \n",
       "01BR015              0.081968               NaN  \n",
       "...                       ...               ...  \n",
       "21BR010             -0.311446          0.602422  \n",
       "22BR005                   NaN          0.456801  \n",
       "22BR006                   NaN          1.094966  \n",
       "CPT000814                 NaN               NaN  \n",
       "CPT001846            0.443203               NaN  \n",
       "\n",
       "[125 rows x 12922 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "proteomics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "8436e461-cf4c-4b0b-9931-6d1cb353a7fc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['01BR001', '01BR008', '01BR009', '01BR010', '01BR015', '01BR017',\n",
       "       '01BR018', '01BR020', '01BR023', '01BR025',\n",
       "       ...\n",
       "       '20BR005', '20BR006', '20BR007', '20BR008', '21BR001', '21BR002',\n",
       "       '21BR010', '22BR005', '22BR006', '604'],\n",
       "      dtype='object', name='Patient_ID', length=133)"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "transcriptomics = en.get_transcriptomics('bcm')\n",
    "transcriptomics.index\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "9e23b96e-e5de-4c16-8b01-cf951c6c9d8a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "âœ… transcriptomics.index æœ‰ï¼Œä½† GDC Sample Sheet æ²’æœ‰çš„ Case ID:\n",
      "set()\n",
      "\n",
      "âœ… GDC Sample Sheet æœ‰ï¼Œä½† transcriptomics.index æ²’æœ‰çš„ Case ID:\n",
      "set()\n",
      "\n",
      "âœ… å…©è€…çš†æœ‰çš„ Case ID:\n",
      "{'06BR014', '11BR044', '09BR001', '01BR033', '05BR045', '05BR042', '13BR009', '11BR032', '18BR009', '05BR029', '05BR044', '11BR053', '18BR003', '14BR005', '03BR006', '11BR036', '11BR016', '11BR012', '01BR032', '05BR055', '11BR038', '03BR010', '11BR054', '11BR027', '06BR005', '15BR003', '05BR009', '100004012', '05BR043', '100003304', '11BR024', '11BR030', '01BR025', '05BR052', '09BR004', '01BR030', '03BR012', '05BR058', '11BR013', '20BR007', '11BR059', '01BR044', '09BR007', '11BR049', '11BR056', '03BR004', '11BR020', '100004028', '1488', '11BR050', '01BR009', '11BR072', '11BR080', '06BR006', '18BR010', '11BR051', '20BR005', '14BR008', '11BR074', '11BR004', '05BR003', '16BR012', '22BR005', '11BR028', '01BR042', '20BR006', '18BR006', '18BR004', '18BR017', '05BR001', '11BR006', '11BR014', '11BR040', '11BR025', '05BR031', '01BR023', '01BR020', '604', '22BR006', '18BR007', '11BR011', '11BR031', '20BR001', '11BR042', '11BR018', '21BR010', '20BR008', '01BR026', '11BR043', '01BR017', '06BR003', '05BR038', '09BR005', '11BR023', '01BR018', '11BR058', '01BR001', '03BR013', '05BR005', '11BR003', '20BR002', '11BR022', '11BR075', '21BR001', '11BR017', '05BR016', '11BR047', '03BR005', '11BR060', '100002921', '05BR004', '21BR002', '01BR015', '01BR027', '11BR069', '11BR010', '01BR040', '03BR002', '11BR009', '01BR010', '18BR002', '11BR019', '18BR016', '11BR055', '01BR043', '14BR014', '01BR008', '01BR031', '18BR019', '11BR073', '11BR015', '05BR026', '05BR051'}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "# 2ï¸âƒ£ è®€å– GDC Sample Sheet\n",
    "gdc_df = pd.read_csv(\"./dataset/TCGA/RNA-Seq/gdc_sample_sheet.2025-02-14.tsv\",sep=\"\\t\")\n",
    "\n",
    "# 3ï¸âƒ£ å–å¾— Case ID é›†åˆ\n",
    "transcriptomics_cases = set(transcriptomics.index)  # transcriptomics çš„ Case ID\n",
    "gdc_cases = set(gdc_df[\"Case ID\"])  # GDC Sample Sheet çš„ Case ID\n",
    "\n",
    "# 4ï¸âƒ£ æ‰¾å‡ºå·®ç•°\n",
    "only_in_transcriptomics = transcriptomics_cases - gdc_cases  # åªåœ¨ transcriptomics å…§\n",
    "only_in_gdc = gdc_cases - transcriptomics_cases  # åªåœ¨ GDC Sample Sheet å…§\n",
    "\n",
    "# 5ï¸âƒ£ å°å‡ºçµæžœ\n",
    "print(f\"âœ… transcriptomics.index æœ‰ï¼Œä½† GDC Sample Sheet æ²’æœ‰çš„ Case ID:\\n{only_in_transcriptomics}\\n\")\n",
    "print(f\"âœ… GDC Sample Sheet æœ‰ï¼Œä½† transcriptomics.index æ²’æœ‰çš„ Case ID:\\n{only_in_gdc}\\n\")\n",
    "\n",
    "# 6ï¸âƒ£ å­˜æˆ CSV æ–¹ä¾¿æŸ¥çœ‹\n",
    "pd.DataFrame({\"Missing in GDC\": list(only_in_transcriptomics)}).to_csv(\"missing_in_gdc.csv\", index=False)\n",
    "pd.DataFrame({\"Missing in transcriptomics\": list(only_in_gdc)}).to_csv(\"missing_in_transcriptomics.csv\", index=False)\n",
    "# å–å¾—å…©è€…çš†æœ‰çš„ Case IDï¼ˆäº¤é›†ï¼‰\n",
    "common_cases = transcriptomics_cases & gdc_cases\n",
    "\n",
    "# å°å‡ºçµæžœ\n",
    "print(f\"âœ… å…©è€…çš†æœ‰çš„ Case ID:\\n{common_cases}\\n\")\n",
    "\n",
    "# å­˜æˆ CSV æ–¹ä¾¿æŸ¥çœ‹\n",
    "pd.DataFrame({\"Common Case IDs\": list(common_cases)}).to_csv(\"common_case_ids.csv\", index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "b6b9a798-f89a-4dd7-a05c-37fc42e3d164",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "133"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(common_cases)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5aee9af7-93a0-4f6f-ab56-39e4ff09854b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.21"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
